File: app/src/main/java/dev/ragnarok/fenrir/view/LoadMoreFooterHelperComment.java
package dev.ragnarok.fenrir.view;

import android.view.View;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.annotation.StringRes;

import dev.ragnarok.fenrir.R;
import dev.ragnarok.fenrir.model.LoadMoreState;

public class LoadMoreFooterHelperComment {

    public Callback callback;
    public Holder holder;
    public int state = LoadMoreState.INVISIBLE;

    public static LoadMoreFooterHelperComment createFrom(View view, Callback callback) {
        LoadMoreFooterHelperComment helper = new LoadMoreFooterHelperComment();
        helper.holder = new Holder(view);
        helper.callback = callback;
        helper.holder.bLoadMore.setOnClickListener(v -> {
            if (callback != null) {
                callback.onLoadMoreClick();
            }
        });
        return helper;
    }

    public void setEndOfListTextRes(@StringRes int res) {
        holder.tvEndOfList.setText(res);
    }

    public void setEndOfListText(String text) {
        holder.tvEndOfList.setText(text);
    }

    public void switchToState(@LoadMoreState int state) {
        this.state = state;
        holder.container.setVisibility(state == LoadMoreState.INVISIBLE ? View.GONE : View.VISIBLE);

        switch (state) {
            case LoadMoreState.LOADING:
                holder.tvEndOfList.setVisibility(View.INVISIBLE);
                holder.bLoadMore.setVisibility(View.INVISIBLE);
                holder.progress.setVisibility(View.VISIBLE);
                break;
            case LoadMoreState.END_OF_LIST:
                holder.tvEndOfList.setVisibility(View.VISIBLE);
                holder.bLoadMore.setVisibility(View.INVISIBLE);
                holder.progress.setVisibility(View.INVISIBLE);
                break;
            case LoadMoreState.CAN_LOAD_MORE:
                holder.tvEndOfList.setVisibility(View.INVISIBLE);
                holder.bLoadMore.setVisibility(View.VISIBLE);
                holder.progress.setVisibility(View.INVISIBLE);
                break;
            case LoadMoreState.INVISIBLE:
                break;
        }
    }

    public interface Callback {
        void onLoadMoreClick();
    }

    public static class Holder {
        public final View container;
        public final ProgressBar progress;
        public final View bLoadMore;
        public final TextView tvEndOfList;

        public Holder(View root) {
            container = root.findViewById(R.id.footer_load_more_root);
            progress = root.findViewById(R.id.footer_load_more_progress);
            bLoadMore = root.findViewById(R.id.footer_load_more_run);
            tvEndOfList = root.findViewById(R.id.footer_load_more_end_of_list);
        }
    }
}

File: app/src/main/java/dev/ragnarok/fenrir/view/LoadMoreFooterHelperComment.kt
package dev.ragnarok.fenrir.view

import android.view.View
import android.widget.ProgressBar
import android.widget.TextView
import androidx.annotation.StringRes
import dev.ragnarok.fenrir.R
import dev.ragnarok.fenrir.model.LoadMoreState

class LoadMoreFooterHelperComment {
    var callback: Callback? = null
    var holder: Holder? = null
    var state = LoadMoreState.INVISIBLE
    fun setEndOfListTextRes(@StringRes res: Int) {
        holder?.tvEndOfList?.setText(res)
    }

    fun setEndOfListText(text: String?) {
        holder?.tvEndOfList?.text = text
    }

    fun switchToState(@LoadMoreState state: Int) {
        this.state = state
        holder?.container?.visibility =
            if (state == LoadMoreState.INVISIBLE) View.GONE else View.VISIBLE
        when (state) {
            LoadMoreState.LOADING -> {
                holder?.tvEndOfList?.visibility = View.INVISIBLE
                holder?.bLoadMore?.visibility = View.INVISIBLE
                holder?.progress?.visibility = View.VISIBLE
            }
            LoadMoreState.END_OF_LIST -> {
                holder?.tvEndOfList?.visibility = View.VISIBLE
                holder?.bLoadMore?.visibility = View.INVISIBLE
                holder?.progress?.visibility = View.INVISIBLE
            }
            LoadMoreState.CAN_LOAD_MORE -> {
                holder?.tvEndOfList?.visibility = View.INVISIBLE
                holder?.bLoadMore?.visibility = View.VISIBLE
                holder?.progress?.visibility = View.INVISIBLE
            }
            LoadMoreState.INVISIBLE -> {}
        }
    }

    interface Callback {
        fun onLoadMoreClick()
    }

    class Holder(root: View) {
        val container: View = root.findViewById(R.id.footer_load_more_root)
        val progress: ProgressBar = root.findViewById(R.id.footer_load_more_progress)
        val bLoadMore: View = root.findViewById(R.id.footer_load_more_run)
        val tvEndOfList: TextView = root.findViewById(R.id.footer_load_more_end_of_list)

    }

    companion object {
        @JvmStatic
        fun createFrom(view: View, callback: Callback?): LoadMoreFooterHelperComment {
            val helper = LoadMoreFooterHelperComment()
            helper.holder = Holder(view)
            helper.callback = callback
            helper.holder?.bLoadMore?.setOnClickListener { callback?.onLoadMoreClick() }
            return helper
        }
    }
}